deicode/optspace.py

```diff
@@ -40,10 +41,12 @@ def __init__(
             N = Features (i.e. OTUs, metabolites)
             M = Samples
 
-        n_components: int
+        n_components: int or {"optspace"}, optional
```

Comment:
Not sure about this - it's generally not good practice to allow for different data types in the same parameter.
Its also misleading - this is to enable automated estimation of rank.  Both options utilize optspace.

I'll recommend to have a separate command but it is up to you.

Reply:
I agree I split the functions into two separate commands.